SwanCustomEnvironments/swancustomenvironments/templates/customenvs.html

```diff
@@ -0,0 +1,160 @@
+{% extends "page.html" %}
+
+{% block stylesheet %}
+{{super()}}
+<style type="text/css">
+/* disable initial hide */
+div#header, div#site {
+    display: block;
+}
+#output-box {
+    display: block;
+    width: 1150px;
+    height: 300px;
+    overflow-y: scroll;
+    overflow-x: hidden;
+    white-space: pre-wrap;
+    text-align: left;
+    border: 1px solid #ccc;
+    background-color: #f7f7f7;
+    padding: 10px;
+    border-radius: 5px;
+}
+#back_button {
+    display: none;
+    text-align: center;
+    background-color: #fc6404;
+    position: absolute;
+    left: 50%;
+    transform: translateX(-50%);
+    color: white;
+    border: none;
+    border-radius: 5px;
+    padding: 10px 20px;
+    font-size: 16px;
+    cursor: pointer;
+    transition: background-color 0.3s ease;
+}
+#next_button {
+    display: none;
+    text-align: center;
+    background-color: #0454a4;
+    position: absolute;
+    left: 50%;
+    transform: translateX(-50%);
+    color: white;
+    border: none;
+    border-radius: 5px;
+    padding: 10px 20px;
+    font-size: 16px;
+    cursor: pointer;
+    transition: background-color 0.3s ease;
+}
+</style>
+{% endblock %}
+
+{% block params %}
+{{super()}}
+data-base-url="{{base_url | urlencode}}"
+{% endblock %}
+
+{% block site %}
+
+<div class="container">
+    <div class="row">
+        <div class="swan-info">
+            <div id="swan-loader">
+                <div class="loader-circle">
+                    <img src="{{hub_prefix}}static/swan/logos/{{swan_logo_filename}}">
+                </div>
+                <div class="loader-line-mask">
+                    <div class="loader-line"></div>
+                </div>
+                <span class="text">Creating environment...</span>
+            </div>
+        </div>
+        <br><br>
+        <div id="output-box"></div>
+        <br>
+        <button id="back_button">Back</button>
+        <button id="next_button">Next</button>
+    </div>
+</div>
+
+{% endblock %}
+
+{% block script %}
+    {{super()}}
+
+<script type = "text/javascript" >
+    
+    var base_url = document.body.getAttribute('data-base-url');
+    
+    const urlParams = new URLSearchParams(window.location.search);
+    var env_name = urlParams.get('env') || "";
+    var repository = urlParams.get('repo') || "";
+    var accpy_version = urlParams.get('accpy') || "";
+    
+    var project_folder = '/';
+    if (repository.startsWith('http')) {
+        // Extract repository name from URL
+        const pattern = /^https?:\/\/[^/]+\/([^/\s]+)\/([^/\s]+)/;
```

Comment:
This pattern and the one of the spawner should be the same no?

Reply:
I was a lot more flexible with these regexes cause they are only intended to extract the path for redirecting the user to the correct URL path.
Example: repo = /eos/user/u/username/folder1/folder2/folder3
URL path = /lab/tree/folder1/folder2/folder3

Reply:
I have a comment that got lost because this file got changed (it's still in the conversation tab).
To me it makes no sense to be parsing the url here and building the path to redirect. The backend already knows where it wrote the files, so it should just return that path and the frontend can use it to redirect.